magento/pwa-studio
# [PWA-1124] Browsing and Viewing Virtual product

## Description

As a Merchant I want to be able to showcase products that don't require delivery on PWA storefront without customization *so that* I decrease the time required to launch a PWA storefront and TTC of it

*Preconditions*
 * Website with two store veiwes (store view 1 and store view 2) created
 * Virtual product created in Admin
 * Virtual product includes localization for each store view
 * Product attribute _Product sub-type_ exists
 * _Product sub-type_ attribute value is set to _Service_ on the virtual product
 * Home Page includes product listing with virtual product
 * User is not logged into storefront account

*Acceptance Criteria*
 * User opens home page and sees virtual product tile that includes:
 ** product image (role: small)
 ** product title
 ** product sub-type (powered by new system attribute)
 ** Product price
 * User opens a category that includes virtual product and sees virtual product tile
 * User navigates to the product details page from the product tile
 * User sees product details page loaded with the product information displayed as designed
 * User can change the language (store view) and sees the product information to be localized
 * User reviews product information
 * User can't add a product to the cart in the scope of this story

*Test Plan*
 * Virtual product should be visible on Home page and on any other CMS pages.
 * Virtual product category should be listed as a circle on home page when PB data is not present.
 * Respective category should be able to list -
 ** Only Virtual Product. Also check sort/filters works.
 ** Combination of Virtual and other Products. Also check sort/filters works.
 * User should be landed on Virtual PDP on accessing from Category list.
 * Verify above scenarios on on second store view and make sure data loads as per configs like language, currency.
 * Verify Offline mode works for Home, Category, PDP which have Virtual product.
 * Run Lighthouse/WPT on virtual product category list and PDP.
 * Run MFTF
 * Check UI scenarios on Chrome, FF, Safari desktop mode. And on Mobile iOS Safari and Android Chrome.
 * {color:#de350b}PENDING - subtype attribute scenarios.{color}

## Related Issue





* [[PWA-1124](https://jira.corp.magento.com/browse/PWA-1124)] 

## Acceptance



### Verification Stakeholders

### Specification



### Verification Steps


1. Make sure you're using a backend that has sub_type defined as described in [this request](https://github.com/PMET-public/module-catalog-sample-data-venia/issues/4)
2. Add `@magento/venia-virtual-products` to `@magento/venia-concept/package.json` && `yarn`
3. Run through test plan and verify created virtual products using this new attribute render as expected

## Screenshots / Screen Captures (if appropriate)

## Checklist

* I have added tests to cover my changes, if necessary.
* I have added translations for new strings, if necessary.
* I have updated the documentation accordingly, if necessary.
